File: Sorting/DutchNationalFlag.java
import java.util.Arrays;

public class DutchNationalFlag {
    public static void main(String[] args) {
        int[] a = {1, -1, 0, 3, 0, 2, 1};
        System.out.println(Arrays.toString(partition(a, 1)));
    }
    public static int[] partition(int[] arr, int pivot) {
        int i = 0;
        int start = 0;
        int end = arr.length - 1;
        while (start < end) {
            if (arr[start] < pivot) {
                swap(arr, i, start);
                i++;
                start++;
            } else if (arr[start] > pivot) {
                swap(arr, start, end);
                end--;
            } else {
                start++;
            }
        }
        return arr;
    }
    public static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}

File: Sorting/DutchNationalFlag.py
def partitionSort(arr):
    if not arr:
        return arr
    n = len(arr)
    pivot = 1
    start, index, end = 0, 0, n - 1

    while index < end:
        if arr[index] < pivot:
            arr[start], arr[index] = arr[index], arr[start]
            start += 1
            index += 1
        elif arr[index] > pivot:
            arr[end], arr[index] = arr[index], arr[end]
            end -= 1
        else:
            index += 1
    return arr

print(partitionSort([1, -1, 0, 3, 0, 2, 1]))